src/components/KYCInfoScreen.tsx

```diff
@@ -1,53 +1,132 @@
 import { QueryParams } from '../schema'
-import { Steps } from '../types'
+import { KycStatus, KycSchemas } from '@fiatconnect/fiatconnect-types'
 import React, { useState } from 'react'
 import { ContentContainer, SectionSubtitle, SectionTitle } from '../styles'
 import {
   providerIdToPrivacyPolicyURL,
   providerIdToProviderName,
 } from '../constants'
-import styled from 'styled-components'
 import KYCInfoFieldSection from './KYCInfoFieldSection'
 import { personalDataAndDocumentsSchemaMetadata } from './kycInfo/PersonalDataAndDocuments'
+import { useFiatConnectConfig } from '../hooks'
+import { addKyc } from '../FiatConnectClient'
+import { KycPending } from './kycStatusScreens/KycPending'
+import { KycDenied } from './kycStatusScreens/KycDenied'
+import { KycExpired } from './kycStatusScreens/KycExpired'
 
 interface Props {
+  kycStatus: KycStatus | undefined
+  setKycStatus: (kycStatus: KycStatus | undefined) => void
   onError: (title: string, message: string) => void
-  onNext: (step: Steps) => void
+  onNext: () => void
   params: QueryParams
 }
 
-export function KYCInfoScreen({ onError, onNext, params }: Props) {
+export function KYCInfoScreen({
+  onError,
+  onNext,
+  params,
+  kycStatus,
+  setKycStatus,
+}: Props) {
   const [submitDisabled, setSubmitDisabled] = useState(true)
   const [kycInfo, setKycInfo] = useState({})
+  const providerName = providerIdToProviderName[params.providerId]
+  const fiatConnectClientConfig = useFiatConnectConfig()
 
-  const onSubmit = () => {
-    // TODO post kyc info, then start polling for kyc status?
+  const setKycInfoWrapper = (
+    newKycInfo: Record<string, string | Record<string, string>>,
+  ) => {
+    setKycInfo({
+      ...kycInfo,
+      ...newKycInfo,
+    })
+  }
+  const onSubmit = async () => {
+    try {
+      if (!params.kycSchema || !fiatConnectClientConfig) {
+        throw new Error()
+      }
+
+      const addKycStatus = await addKyc(
+        {
+          kycSchemaName: params.kycSchema,
+          data: kycInfo as KycSchemas[typeof params.kycSchema],
+        },
+        fiatConnectClientConfig,
+      )
+      setKycStatus(addKycStatus)
+    } catch (error) {
+      onError(
+        `There was an error while submitting your KYC information to ${providerName}`,
+        'This may be due to a problem with the provider.',
+      )
+    }
+  }
+
+  if (kycStatus === KycStatus.KycPending) {
+    if (!params.kycSchema) return
```

Comment:
these checks make me a little nervous. maybe the props should be of type QueryParams & {kycSchema: KycSchema} ...